### PR TITLE
fix: str concat in error call

### PR DESCRIPTION
**Pull request by cache-missing**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tokers** [comment]
@cache-missing The PR title is not semantic, please fix it, see https://github.com/zeke/semantic-pull-requests.

File: apisix/plugin.lua [hunk 1]
```diff
@@ -264,7 +264,7 @@ function _M.load(config)
         local_conf, err = core.config.local_conf(true)
         if not local_conf then
             -- the error is unrecoverable, so we need to raise it
-            error("failed to load the configuration file: ", err)
+            error("failed to load the configuration file: " .. err)
```
Comment (moonming) [inline apisix/plugin.lua:267]:
```suggestion
            core.log.error("failed to load the configuration file: ", err)
```

Comment (spacewander) [reply apisix/plugin.lua:267]: We need to raise error here as it is unrecoverable. Otherwise, APISIX is still in trouble.

**Review by spacewander — APPROVED** [review]

File: apisix/plugin.lua (hunk 1 above)
Comment (moonming) [reply apisix/plugin.lua:267]: got it

**Review by moonming — APPROVED** [review]

**Comment by lijing-21** [comment]
Hi @cache-missing  , thank you for your contribution!
Now you are eligible to receive the very special [Contributor T-shirt](https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#contributor-t-shirt)！
Just fill out the [Contributors Submissions form](https://docs.google.com/forms/d/e/1FAIpQLSdXEpCs60UK49UlOGdBCQSXr7DYz3enyT4GJPKrYQmYfVLPKQ/viewform)~

**Pull request merged**